### DISCUSSION TITLE
Is there a way to convert a script to a standalone (single file) executable?

### DISCUSSION THREAD

**Discussion by misha-franz** [aardappel/lobster#311] [Q&A]
Is there a way to convert a script to a standalone (single file) executable?

**Comment by aardappel** [comment]
Yes, it's all in the documentation: https://aardappel.github.io/lobster/implementation.html
In particular the `Compiling Lobster code to C++` part.

**Comment by misha-franz** [comment]
I likely did not explain myself well enough.  My point it to use Lobster do quickly develop simple scripts and share them with other people as executable.  I have this kind of experience with both Tcl/Tk and Lua.  In both cases I think the idea is to attach a script (or a precompiled bytecode) to the interpreter.  Then when the interpreter starts, it first checks the app name and if it is different from default, it check after the end of the file and looking for a script to execute.  This seems only relevant for Windows.

**Reply by aardappel** [reply]
The answer is the same as the above. That is the only procedure we currently have to convert to executable..

If you want an easier way to distribute, you can convert the source to a .pak file (which contains bytecode and other needed files), which you can bundle with lobster.exe, thus just needs 2 files.
